src/task_queue/database_tasks/translate_contents.py

```diff
@@ -0,0 +1,176 @@
+                    target_language_code=lang
+                ).first()
+                
+                if not translation_cache:
+                    try:
+                        translated_text = self.translate_text(text, lang)
+                        if not translated_text:
+                            return None
+                        translated_cache = self.cache_translation(text_hash, source_language_code, lang, translated_text)
+                        return translated_cache.translated_text
+                    
+                    except TranslationError as e:
+                        logger.log(f"Translation process encountered an error: {e}")
+                        return None
+                
+                else:
+                    logger.log("Translation already exists in cache")
+                    return translation_cache.translated_text
+        except Exception as e:
+            logger.log(f"Error occurred while translating field: {e}")
+            return None
+    
+    def translate_model_instance(self, instance, fields_to_translate):
+        try:
+            for field in fields_to_translate:
+                field_value = getattr(instance, field)
+                field_type = instance._meta.get_field(field)
+                
+                if isinstance(field_type, JSONField):
+                    if isinstance(field_value, dict):
+                        for key, value in field_value.items():
+                            self.translate_field(value, SOURCE_LANGUAGE_CODE)
+                else:
+                    self.translate_field(field_value, SOURCE_LANGUAGE_CODE)
+            return True
+        except Exception as e:
+            logger.log(f"Error occurred while translating model instance: {e}")
+            return False
+    
+    def translate_menu_model(self, model):
+        try:
+            instances = model.objects.all()
+            exclude_fields = ["link", "is_published", "id", "is_link_external"]
+            for instance in instances:
+                
+                content = JsonTranslator(instance.content, exclude_fields).get_flattened_dict()
+                footer_content = JsonTranslator(instance.footer_content, exclude_fields).get_flattened_dict()
+                
+                logger.log(f"Translating Nav contents for menu: {instance.name}")
+                for key, value in content.items():
+                    self.translate_field(value, SOURCE_LANGUAGE_CODE)
+                
+                logger.log(f"Translating Footer contents for menu: {instance.name}")
+                for key, value in footer_content.items():
+                    self.translate_field(value, SOURCE_LANGUAGE_CODE)
+                    
+            return True
+        except Exception as e:
+            logger.log(f"Error occurred while translating menu model: {e}")
+            return None
+        
+    def get_valid_instances(self, model):
```

Comment:
what is the use of this?

Reply:
I didn't want us translating stale content, which is why I added these filters.